File: Agregacion/E-7/Python/Estudiante.py

class Estudiante:
    def __init__(self, nombre, carrera, semestre):
        self.nombre = nombre
        self.carrera = carrera
        self.semestre = semestre
    
    
    def get_nombre(self):
        return self.nombre
    
    def set_nombre(self, nombre):
        self.nombre = nombre
    
    def get_carrera(self):
        return self.carrera
    
    def set_carrera(self, carrera):
        self.carrera = carrera
    
    def get_semestre(self):
        return self.semestre
    
    def set_semestre(self, semestre):
        self.semestre = semestre
    
    
    def mostrar_info(self):
        print(f"Nombre: {self.nombre}, Carrera: {self.carrera}, Semestre: {self.semestre}")


class Universidad:
    def __init__(self, nombre):
        self.nombre = nombre
        self.estudiantes = []
    
    
    def get_nombre(self):
        return self.nombre
    
    def set_nombre(self, nombre):
        self.nombre = nombre
    
    def get_estudiantes(self):
        return self.estudiantes
    
    
    def agregar_estudiante(self, estudiante):
        self.estudiantes.append(estudiante)
    
    
    def mostrar_universidad(self):
        print(f"Universidad: {self.nombre}")
        print("Estudiantes:")
        for estudiante in self.estudiantes:
            estudiante.mostrar_info()


es1 = Estudiante("Luis Pérez", "Ingeniería Ambiental", 5)
es2 = Estudiante("Neytan García", "Medicina", 7)
es3 = Estudiante("Fernando López", "Derecho", 1)


miu = Universidad("UMSS")


miu.agregar_estudiante(es1)
miu.agregar_estudiante(es2)
miu.agregar_estudiante(es3)


miu.mostrar_universidad()

File: Agregacion/E-7/Java-Eclipse/src/agregacion/Estudiante.java
package agregacion;
import java.util.ArrayList;
import java.util.List;

public class Estudiante {
	private String nombre;
    private String carrera;
    private int semestre;

    // Constructor
    public Estudiante(String nombre, String carrera, int semestre) {
        this.nombre = nombre;
        this.carrera = carrera;
        this.semestre = semestre;
    }

    // Getters y Setters
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getCarrera() {
        return carrera;
    }

    public void setCarrera(String carrera) {
        this.carrera = carrera;
    }

    public int getSemestre() {
        return semestre;
    }

    public void setSemestre(int semestre) {
        this.semestre = semestre;
    }

    // Método para mostrar información
    public void mostrarInfo() {
        System.out.println("Nombre: " + nombre + ", Carrera: " + carrera + ", Semestre: " + semestre);
    }
}
class Universidad {
    private String nombre;
    private List<Estudiante> estudiantes;

    // Constructor
    public Universidad(String nombre) {
        this.nombre = nombre;
        this.estudiantes = new ArrayList<>();
    }

    // Getters y Setters
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public List<Estudiante> getEstudiantes() {
        return estudiantes;
    }

    // Método para agregar estudiante
    public void agregarEstudiante(Estudiante estudiante) {
        estudiantes.add(estudiante);
    }

    // Método para mostrar información de la universidad
    public void mostrarUniversidad() {
        System.out.println("Universidad: " + nombre);
        System.out.println("Estudiantes:");
        for (Estudiante estudiante : estudiantes) {
            estudiante.mostrarInfo();
        }
    }
}
